beacon-chain/forkchoice/protoarray/store.go

```diff
@@ -559,9 +560,8 @@ func (s *Store) insert(ctx context.Context,
 		s.receivedBlocksLastEpoch[slot%params.BeaconConfig().SlotsPerEpoch] = slot
 	}
 	// Update highest slot tracking.
-	if slot > s.highestReceivedSlot {
-		s.highestReceivedSlot = slot
-		s.highestReceivedRoot = root
+	if slot > s.highestReceivedNode.slot {
```

Comment:
In extreme cases of latency I think highestReceivedNode could have been deleted by pruning in finalization, so a nil check here is useful

Reply:
Makes sense. Thanks!